### DISCUSSION TITLE
Precedent for configuring user settings?

### DISCUSSION THREAD

**Discussion by MathewBlock** [ScoopInstaller/Scoop#4605] [Help and Questions] [answered]
I've been thinking about the issue of applications that have built-in auto updaters. I was wondering if there is any precedent with other apps registered here where there is some mechanism for configuring user settings. For instance, if the auto updating application has a configuration setting to disable auto update is that something we'd want Scoop to configure for the user (either automatically or optionally)?

**Comment by HUMORCE** [comment]
the applications that can be configured to disable built-in updates are mostly disabled. 

for use built-in updater: `scoop hold <app>` and turn on app's built-in updater.

**Reply by MathewBlock** [reply]
Just to clarify I was more asking this question as a developer, rather than a user. Specifically, Discord, which has built in updating and I believe a way to disable that built-in updating via user settings. 

Using `scoop hold` is one approach (letting the app do its thing) but I'd like to explore doing it the other way around, having scoop configure those settings on install (to tell the app not to update), then use Scoop to update. 

You mention that applications that can be configured to disable built-in updates are mostly disabled. I'm interesting in learning the mechanisms used to implement that, so I can extend other manifests to do the same. Could you perhaps share the names of some apps that already do this, so I can use them as examples?

**Comment by chawyehsu** [comment, marked as answer]
You can use `isntaller`/`pre_install`/`post_install` to turn off apps' built-in updates if they have the settings allowing to do that.

https://github.com/ScoopInstaller/Extras/blob/5b3dc7f7b5b5dd87529312b878769491bd5ef22f/bucket/potplayer.json#L44
https://github.com/ScoopInstaller/Extras/blob/849596cfb144236c2d76ea7c2d5c7071d2313f70/bucket/free-manga-downloader2.json#L17

**Reply by MathewBlock** [reply]
Thanks, that's what I was looking for. 

For my situation with Discord it looks a bit more complicated since the settings file I need to modify exists in `%appdata%` and not one of the application directories. Has there been any other Scoop app that has attempted to redirect or modify the contents of a file in `%appdata%`? Or perhaps provided a script that can be executed by the user that makes the necessary changes?

**Reply by rashil2000** [reply]
If the config file exists in %appdata%, I would not recommend modifying it, because it might contain user's settings for installations done outside of Scoop. We only set/modify config files that exist in the installation folder itself.

You can add a note in the manifest to ask the user to manually disable it (or provide a script that they can run manually).

**Reply by MathewBlock** [reply]
Thanks, that's the confirmation I was looking for.

Since the application in question doesn't have any UI for disabling auto updates, I think an optional powershell script that makes the necessary change is probably best then.
